File: Backend/service/MealPlanService.java
package com.example.demo.service;

import com.example.demo.dto.MealPlanDTO;
import com.example.demo.dto.NoteDTO;

import java.time.LocalDate;
import java.util.List;

public interface MealPlanService {
    MealPlanDTO getMealPlanForDate(String userEmail, LocalDate date);
    List<MealPlanDTO> getAllMealPlans(String userEmail);
    void addRecipeToMealPlan(String userEmail, LocalDate date, Long recipeId);
    void removeRecipeFromMealPlan(String userEmail, LocalDate date, Long recipeId);
    NoteDTO addNoteToMealPlan(String userEmail, LocalDate date, NoteDTO noteDTO);
    void deleteNoteFromMealPlan(String userEmail, LocalDate date, Long noteId);
}

File: Frontend_Mobile/utils/MealPlanService.kt
package com.example.recipeapp.utils

import retrofit2.Call
import retrofit2.http.*

interface MealPlanService {

    @GET("api/mealplans")
    fun getAllMealPlans(): Call<List<MealPlanDTO>>

    @GET("api/mealplans/{date}")
    fun getMealPlanForDate(
        @Path("date") date: String // ISO date string, e.g., "2024-12-19"
    ): Call<MealPlanDTO>

    @POST("api/mealplans/{date}/recipes/{recipeId}")
    fun addRecipeToMealPlan(
        @Path("date") date: String,
        @Path("recipeId") recipeId: Long
    ): Call<Void>

    @DELETE("api/mealplans/{date}/recipes/{recipeId}")
    fun removeRecipeFromMealPlan(
        @Path("date") date: String,
        @Path("recipeId") recipeId: Long
    ): Call<Void>

    @POST("api/mealplans/{date}/notes")
    fun addNoteToMealPlan(
        @Path("date") date: String,
        @Body noteDTO: NoteDTO
    ): Call<NoteDTO>

    @DELETE("api/mealplans/{date}/notes/{noteId}")
    fun deleteNoteFromMealPlan(
        @Path("date") date: String,
        @Path("noteId") noteId: Long
    ): Call<Void>
}
